lib/src/model/robotoff_nutrient_extraction.dart

```diff
@@ -0,0 +1,201 @@
+      _$NutrientEntityFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientEntityToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotationData {
+  final String? unit;
+  final String? value;
+
+  const NutrientAnnotationData({
+    this.unit,
+    this.value,
+  });
+
+  factory NutrientAnnotationData.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationDataToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotation {
+  final Map<String, NutrientAnnotationData>? nutrients;
+  @JsonKey(name: 'serving_size')
+  final String? servingSize;
+  @JsonKey(name: 'nutrition_data_per')
+  final String? nutritionDataPer;
+
+  const NutrientAnnotation({
+    this.nutrients,
+    this.servingSize,
+    this.nutritionDataPer,
+  });
+
+  factory NutrientAnnotation.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationToJson(this);
+}
+
+@JsonSerializable()
+class NutrientDataWrapper {
+  final Map<String, List<NutrientEntity>>? entities;
+  final Map<String, NutrientEntity>? nutrients;
+  final NutrientAnnotation? annotation;
+  @JsonKey(name: 'was_updated')
+  final bool? wasUpdated;
+
+  const NutrientDataWrapper({
+    this.entities,
+    this.nutrients,
+    this.annotation,
+    this.wasUpdated,
+  });
+
+  factory NutrientDataWrapper.fromJson(Map<String, dynamic> json) =>
+      _$NutrientDataWrapperFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientDataWrapperToJson(this);
+}
+
+@JsonSerializable()
+class NutrientExtractionInsight extends JsonObject {
+  @JsonKey(name: 'id')
+  final String? insightId;
+  final String? barcode;
+  final String? type;
+  final NutrientDataWrapper? data;
+  final String? timestamp;
+  @JsonKey(name: 'completed_at')
+  final String? completedAt;
+  final int? annotation;
+  @JsonKey(name: 'annotated_result')
+  final int? annotatedResult;
+  @JsonKey(name: 'n_votes')
+  final int? nVotes;
+  final String? username;
+  final List<String>? countries;
+  final List<String>? brands;
+  @JsonKey(name: 'process_after')
+  final String? processAfter;
+  @JsonKey(name: 'value_tag')
+  final String? valueTag;
+  final String? value;
+  @JsonKey(name: 'source_image')
+  final String? sourceImage;
+  @JsonKey(name: 'automatic_processing')
+  final bool? automaticProcessing;
+  @JsonKey(name: 'server_type')
+  final String? serverType;
+  @JsonKey(name: 'unique_scans_n')
+  final int? uniqueScansN;
+  @JsonKey(name: 'reserved_barcode')
+  final bool? reservedBarcode;
+  final String? predictor;
+  @JsonKey(name: 'predictor_version')
+  final String? predictorVersion;
+  final List<String>? campaign;
+  final double? confidence;
+  @JsonKey(name: 'bounding_box')
+  final dynamic boundingBox;
+
+  const NutrientExtractionInsight({
+    this.insightId,
+    this.barcode,
```

Comment:
I'm not sure using `final` fields is very helpful.
That forces us to maintain the list of fields in the constructor.